pinot-common/src/main/java/org/apache/pinot/common/data/FieldSpec.java

```diff
@@ -98,15 +102,21 @@ public FieldSpec(String name, DataType dataType, boolean isSingleValueField) {
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, @Nullable Object defaultNullValue) {
     this(name, dataType, isSingleValueField, DEFAULT_MAX_LENGTH, defaultNullValue);
   }
-
+  
   public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
       @Nullable Object defaultNullValue) {
+    this(name, dataType, isSingleValueField, maxLength, defaultNullValue, null);
+  }
+  
+  public FieldSpec(String name, DataType dataType, boolean isSingleValueField, int maxLength,
```

Comment:
What's the issue with extending DataType to capture the object type?

Reply:
yes, that was the initial solution. But wanted to keep the io layer (reader and writers) separate from the logical object layer. In other words, get<DataType> is supported by all the readers and writers.

Also, there are too many switch(dataType) in the code which needs to handle various these DataTypes and only thing we can do there is handle them as string or bytes without much value.